platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/fingerprint/ConfigurationCacheFingerprintWriter.kt

```diff
@@ -886,9 +886,4 @@ class ConfigurationCacheFingerprintWriter(
 
 
 internal
-fun jvmFingerprint() = String.format(
-    "%s|%s|%s",
-    System.getProperty("java.vm.name"),
-    System.getProperty("java.vm.vendor"),
-    System.getProperty("java.vm.version")
-)
+fun jvmFingerprint() = "${System.getProperty("java.vm.name")}|${System.getProperty("java.vm.vendor")}|${System.getProperty("java.vm.version")}"
```

Comment:
🤔 Thank looks worse to me than the previous variant.
Consider something like
```kotlin
listOf(
  System.getProperty("java.vm.name"),
  System.getProperty("java.vm.vendor"),
  System.getProperty("java.vm.version"),
).joinToString("|")
```

Reply:
Much nicer indeed, ok!

Reply:
FYI, the suggested code above is incorrect, took some doing to link it to failing CC tests... Correct version:
```
listOf(
    System.getProperty("java.vm.name"),
    System.getProperty("java.vm.vendor"),
    System.getProperty("java.vm.version")
).joinToString (separator = "|")
```